lib/lwm2m_client/CMakeLists.txt

```
@@ -0,0 +1,52 @@
+#
+# Copyright (c) 2019 Nordic Semiconductor
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+zephyr_include_directories(include)
+zephyr_library()
+
+file(GLOB FOTA_SRC "${CMAKE_CURRENT_LIST_DIR}/fota/*.c")
+file(GLOB LWM2M_SRC "${CMAKE_CURRENT_LIST_DIR}/lwm2m/*.c")
+file(GLOB UI_SRC "${CMAKE_CURRENT_LIST_DIR}/ui/*.c")
+
+if(NOT CONFIG_LWM2M_LOCATION_OBJ_SUPPORT)
+	list(REMOVE_ITEM LWM2M_SRC "${CMAKE_CURRENT_SOURCE_DIR}/lwm2m/lwm2m_location.c")
+endif()
+if(NOT CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT)
+	list(REMOVE_ITEM LWM2M_SRC "${CMAKE_CURRENT_SOURCE_DIR}/lwm2m/lwm2m_firmware.c")
+endif()
+if(NOT CONFIG_LWM2M_CONN_MON_OBJ_SUPPORT)
+	list(REMOVE_ITEM LWM2M_SRC "${CMAKE_CURRENT_SOURCE_DIR}/lwm2m/lwm2m_connmon.c")
+endif()
+if(NOT CONFIG_LWM2M_IPSO_LIGHT_CONTROL)
+	list(REMOVE_ITEM LWM2M_SRC "${CMAKE_CURRENT_SOURCE_DIR}/lwm2m/lwm2m_light_control.c")
+endif()
+if(NOT CONFIG_LWM2M_IPSO_TEMP_SENSOR)
+	list(REMOVE_ITEM LWM2M_SRC "${CMAKE_CURRENT_SOURCE_DIR}/lwm2m/lwm2m_temp.c")
+endif()
+if(NOT CONFIG_UI_BUZZER)
+	list(REMOVE_ITEM LWM2M_SRC "${CMAKE_CURRENT_SOURCE_DIR}/lwm2m/lwm2m_buzzer.c")
+endif()
+if(NOT CONFIG_UI_BUTTON)
+	list(REMOVE_ITEM LWM2M_SRC "${CMAKE_CURRENT_SOURCE_DIR}/lwm2m/lwm2m_button.c")
+endif()
+if(NOT CONFIG_LWM2M_IPSO_ACCELEROMETER)
+	list(REMOVE_ITEM LWM2M_SRC "${CMAKE_CURRENT_SOURCE_DIR}/lwm2m/lwm2m_accelerometer.c")
+endif()
```

```suggestion
zephyr_library_sources_ifdef(CONFIG_LWM2M_CONN_MON_OBJ_SUPPORT lwm2m/lwm2m_connmon.c)
zephyr_library_sources_ifdef(CONFIG_LWM2M_IPSO_LIGHT_CONTROL   lwm2m/lwm2m_light_control.c)
zephyr_library_sources_ifdef(CONFIG_LWM2M_IPSO_TEMP_SENSOR     lwm2m/lwm2m_temp.c)
zephyr_library_sources_ifdef(CONFIG_UI_BUZZER                  lwm2m/lwm2m_buzzer.c)
zephyr_library_sources_ifdef(CONFIG_UI_BUTTON                  lwm2m/lwm2m_button.c)
zephyr_library_sources_ifdef(CONFIG_LWM2M_IPSO_ACCELEROMETER   lwm2m/lwm2m_accelerometer.c)
```

Remember to add any non-covered files, that are no longer added when globbing is not used.

---

Fixed.